hugo/content/docs/tutorials/wine.md

```diff
@@ -0,0 +1,256 @@
+
+## Environment Setup
+
+We will be using python and sklearn to train our classifier.
+
+We recommend that you first install a python virtual environment.
+```
+$ python -m venv .env
+$ source .env/bin/activate
+```
+
+We can then install the python packages we need.
+``` 
+$ pip install scikit-learn==0.23.2
+```
+
+We will also need to install `dud` which requires `go` to be in your PATH.
+```
+$ go get github.com/kevin-hanselman/dud
+```
+
+You can verify that `dud` installed correctly by outputting the CLI help docs.
+```
+$ dud -h
+Dud is a tool for storing, versioning, and reproducing large files alongside
+source code.
+
+Usage:
+  dud [command]
+
+Available Commands:
+  checkout    Load committed artifacts from the cache
+  checksum    Checksum files or bytes from STDIN
+  commit      Save artifacts to the cache and record their checksums
+  completion  Generate shell completion script
+  config      Print or modify fields in the config file
+  fetch       Fetch committed artifacts to the default remote cache
+  graph       Print stage graph in graphviz DOT format
+  help        Help about any command
+  init        Initialize a Dud project
+  push        Push committed artifacts to the default remote cache
+  run         Run stages or pipelines
+  stage       Commands for interacting with stages and the index
+  status      Print the state of one or more stages
+  version     Print version and exit
+
+Flags:
+  -h, --help      help for dud
+      --profile   enable profiling
+      --trace     enable tracing
+
+Use "dud [command] --help" for more information about a command.
+```
+
+Before we add any data, let's initialize a `dud` repo.
+```
+$ dud init
+Dud project initialized.
+See .dud/config.yaml and .dud/rclone.conf to customize the project.
+```
+
+This creates a `.dud` folder in the current folder and populates it with some config files that have sensible defaults. A `.dud/cache` folder is also created, but it's empty for now. 
+
+Let's verify that the above is true.
+```
+$ tree .dud
+.dud
+├── cache
+├── config.yaml
+└── rclone.conf
+```
+
+Now we're ready to add some data.
+
+## Adding Data
+
+We will be using the "Wine Quality Data Set" from the UCI Machine Learning Repository, which we can download easily with `wget`.
+
+The command to download the dataset is shown below.
+```
+$ wget -q -r -np -nd -A csv https://archive.ics.uci.edu/ml/machine-learning-databases/wine-quality/ -P data
+```
+If you're unfamiliar with `wget`, all you need to know is that this downloads a couple CSVs and saves them in the `data` folder.
+
+We can then create a `dud` stage to run the above command for us. A stage is a group of file or directories (a file or directory is known to `dud` as an artifact), or an _operation_ on a group of artifacts. A stage is defined by a YAML file, and can be tracked with source control. 
```

Comment:
"group of files", plural.

Maybe promote the parenthetical note to its own sentence? Understanding stages and artifacts is worth more explicitly calling out. Maybe start with what a stage is, show `dud stage gen`, introduce the YAML, note that the `outputs` are artifacts, and then describe what artifacts are. Feel free to steal language from the Terminology section currently in the README (I'm hoping to move it soon).